src/components/20-molecules/file-upload/index.js

```diff
@@ -0,0 +1,426 @@
+    this.allDroppedFiles = 0;
+    this.isFileMaxReached = false;
+
+    this.globalErrorMessage = '';
+    this.addMoreInputFile = '';
+    this.showAddMoreInputFile = '';
+  }
+
+  handleAddMoreInputClick() {
+    /* trigger inputFile component after clicking the "fake" inputFile (addMoreInput) */
+    this.inputFile.querySelector('input').click();
+  }
+
+  handleInputFileChange(e) {
+    this.addFiles(e.target.files);
+  }
+
+  handleDropZoneDragover(e) {
+    /* prevent default browser behavior to execute the link that comes with the event */
+    e.preventDefault();
+    if (!this.isFileMaxReached) {
+      e.dataTransfer.dropEffect = 'copy';
+      this.dropZone.classList.add('m-file-upload__dropzone_dragover');
+    }
+  }
+
+  handleDropZoneDragleave() {
+    this.dropZone.classList.remove('m-file-upload__dropzone_dragover');
+  }
+
+  handleDropZoneDrop(e) {
+    /* prevent browser to display the file fullscreen */
+    e.preventDefault();
+
+    const files = [...e.dataTransfer.files].filter(
+      file => ACCEPTED_FILE_TYPES.indexOf(file.type) > -1
+    );
+
+    this.dropZone.classList.remove('m-file-upload__dropzone_dragover');
+    if (files.length > 0 && !this.isFileMaxReached) {
+      this.addFiles(files);
+    }
+  }
+
+  handleFileDeletion(index) {
+    let clonedFiles = [];
+
+    this.allDroppedFiles =
+      (this.files.length === this.maxNumberOfFiles
+        ? this.maxNumberOfFiles
+        : this.allDroppedFiles) - 1;
+
+    this.globalErrorMessage = '';
+
+    if (index >= this.files.length) {
+      /* wrong file */
+      clonedFiles = [...this.faultyFiles];
+      clonedFiles.splice(index - this.files.length, 1);
+      this.faultyFiles = clonedFiles;
+      this.allDroppedFiles += 1;
+    } else {
+      /* valid file */
+      clonedFiles = [...this.files];
+      clonedFiles.splice(index, 1);
+      this.files = clonedFiles;
+      if (this.files.length + 1 === this.maxNumberOfFiles) {
+        this.showAddMoreInputFile = true;
+      }
+    }
+
+    this.handleMaxNumberOfFiles();
+
+    this.sizeOfAllFilesByte -= this.allFiles[index].size;
+
+    if (this.files.length + this.faultyFiles.length === 0) {
+      this.showFileOverview = false;
+      this.sizeOfAllFilesByte = 0;
+      this.allDroppedFiles = 0;
+      this.files = [];
+      this.allFiles = [];
+      this.faultyFiles = [];
+      this.showAddMoreInputFile = false;
+    }
+    this.requestUpdate();
+  }
+
+  async addFiles(droppedFiles) {
+    this.showAddMoreInputFile = true;
+    this.globalErrorMessage = '';
+    this.allDroppedFiles += droppedFiles.length;
+    const filesLeftOver = Math.max(
+      this.maxNumberOfFiles - this.files.length,
+      0
+    );
+
+    /* droppedFiles is not an Array, therefore we can not use slice */
+    const slicedFiles = Array.prototype.slice.call(
+      droppedFiles,
+      0,
+      filesLeftOver
+    );
+
+    const pdfs = [...slicedFiles].filter(file => file.type.indexOf('pdf') > -1);
```

Comment:
this is a requirement for this file upload. only pdfs allowed so far

Reply:
should be documented well on the Readme and also maybe a hint that if someone needs another file type to create a feature request.

If we support the whole world from begging we will have lots of issues

Reply:
Agree

Reply:
Yes, for now. If in the future, docx and all the others also get allowed, we will have to do a big refactoring. And the poor pig who wants to implement this feature will probably end up debugging, since the pdf stuff is hardcoded in this method.

We can still restrict files to only images and pdf, but i would use another filter method here. Instead of filtering "all pdfs", I would filter all files not being "png, jpg, gif, ...". This would be clean architecture that allows for easy future refactorings. If we do not do that, some poor soul will end up debugging because the code crashes when it tries to compress a docx (or whatever) file. This here is error prone.

Reply:
done